autosklearn/experimental/askl2.py

```diff
@@ -25,15 +25,16 @@
     m = hashlib.md5()
     m.update(fh.read().encode('utf8'))
 training_data_hash = m.hexdigest()[:10]
-sklearn_version = sklearn.__version__
-autosklearn_version = autosklearn.__version__
-selector_file = pathlib.Path(
-    os.environ.get(
-        'XDG_CACHE_HOME',
-        '~/.cache/auto-sklearn/askl2_selector_%s_%s_%s.pkl'
-        % (autosklearn_version, sklearn_version, training_data_hash),
-    )
-).expanduser()
+selector_filename = "askl2_selector_%s_%s_%s.pkl" % (
+    autosklearn.__version__,
+    sklearn.__version__,
+    training_data_hash
+)
+selector_directory = os.environ.get('XDG_CACHE_HOME')
+if selector_directory is None or not os.access(selector_directory, os.W_OK):
```

Comment:
Hey, what's the reason to check if that directory is writable? Also, wouldn't you also have to check the selector directory below?

I think if 'XDG_CACHE_HOME' is set but not writable or the fallback is not writable either, we should not save the selector. I'm also wondering whether we should assume the existence of a directory called `.cache` or whether we should just write into `.auto-sklearn`?

Reply:
Yeah, writting to `.auto-sklearn` sounds better. Then I wonder, why using `XDG_CACHE_HOME`. Do you think it is more straight forward to:

1. Default to `<HOME>/.auto-sklearn`
2. If the user specified `AUTO_SKLEARN_SELECTO_PATH` we prioritize that?

Reply:
The idea of using `XDG_CACHE_HOME` is to follow the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) which a few packages under linux follow. To summarize, I think it would be best to either write in there or not at all to have consistent behavior.